### PR TITLE
Hw4 task3

### PR DESCRIPTION
**Pull request by E1-Bart0**

### CODE REVIEW

File: test/hw4/test_task3.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from All_home_works.hw4.task3 import my_precious_logger
+
+ERROR = "error: 404"
+TEXT = "OK: 200"
```
Comment (Cjkjvfnby) [inline test/hw4/test_task3.py]: It is better to inline these variables.

File: test/hw4/test_task3.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+def test_my_precious_logger__error_msg__test_msg_in_stderr(capsys):
```
Comment (Cjkjvfnby) [inline test/hw4/test_task3.py]: It's ok to join this test with test_my_precious_logger__error_msg__test_none_in_stdout

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Let's try to find a bit more cases we could test here.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
https://github.com/E1-Bart0/homework-repo/pull/19#pullrequestreview-649813423

File: test/hw4/test_task3.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+def test_my_precious_logger__ok__msg__test_msg_in_stdout(capsys):
+    text = "OK: 200"
+    my_precious_logger(text=text)
+    captured = capsys.readouterr()
+    assert captured.out == text
+    assert not captured.err
+
+
+def test_my_precious_logger__ok__msg_start_as_ordinary_msg__test_msg_in_stdout(capsys):
```
Comment (Cjkjvfnby) [inline test/hw4/test_task3.py]: What is the difference between Ok and ordinary cases?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Let's add one more positive test.   hint: the key is in the task requirements.

File: test/hw4/test_task3.py (hunk 3 above)
Comment (E1-Bart0) [reply test/hw4/test_task3.py]: I think it's not actually needed to test.

File: test/hw4/test_task3.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+def test_my_precious_logger__error_msg__test_msg_in_stderr(capsys):
+    error = "error: 404"
+    my_precious_logger(text=error)
+    captured = capsys.readouterr()
+    assert captured.err == error
+    assert not captured.out
+
+
+@pytest.mark.parametrize("input_data", ["Error: 404", "error 404"])
```
Comment (Cjkjvfnby) [inline test/hw4/test_task3.py]: Almost there, need to find one more combination with `error`.  There is a keyword for it in the requirements. 

Also, this test helped me to find a bug in your code.  One of the `input_data` should be in stderr.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: test/hw4/test_task3.py (hunk 4 above)
Comment (E1-Bart0) [reply test/hw4/test_task3.py]: Task description:

> if line starts with "error"

My mistake is: 'error:'

And next combination is: 'errorTEXT' or 'errorerror'?

Comment (Cjkjvfnby) [reply test/hw4/test_task3.py]: > And next combination is: 'errorTEXT' or 'errorerror'?

No. 

This is in that phrase `if line starts with "error"`  and it is not `error`.

Comment (E1-Bart0) [reply test/hw4/test_task3.py]: We are not considering bytes string? I do not know.

Comment (Cjkjvfnby) [reply test/hw4/test_task3.py]: > We are not considering bytes string? 

This looks like a randomly dropped suggestion.  I don't see any hints in the phrase `if a line starts with "error"` which could lead to bytes string.   

There is a technique named brainstorming when you throw some raw ideas. But brainstorming is not used in the current context. In teamwork, the fast and random answer gives more problems than it solves. Just simple `I do not know` would be much better. 

You need to find some cases that could violate this requirement:
```
if line starts with "error"
```

Comment (E1-Bart0) [reply test/hw4/test_task3.py]: I'm sorry, but I do not know...

Comment (Cjkjvfnby) [reply test/hw4/test_task3.py]: You have tested `if`, you have tested `error` what else has left?

Comment (E1-Bart0) [reply test/hw4/test_task3.py]: if "error" in line, but not line startswith "error"

Comment (Cjkjvfnby) [reply test/hw4/test_task3.py]: Exactly!

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**